public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/NestedRows.tsx

```diff
@@ -0,0 +1,195 @@
+  useEffect(() => {
+    const selectedRow = Object.keys(selectedRows).map((rowId) => selectedRows[rowId])[0];
+    const isSelectedResourceGroup =
+      selectedRow && selectedRow.resourceGroupName && row.name === selectedRow.resourceGroupName;
+    if (isSelectedResourceGroup) {
+      setOpenStatus('open');
+    }
+  }, [selectedRows, row]);
+
+  return (
+    <>
+      <tr className={cx(styles.row, isDisabled && styles.disabledRow)} key={row.id}>
+        <td className={styles.cell}>
+          <NestedEntry
+            level={level}
+            isSelected={isSelected}
+            isDisabled={isDisabled}
+            isOpen={isOpen}
+            entry={row}
+            onToggleCollapse={onRowToggleCollapse}
+            onSelectedChange={onRowSelectedChange}
+          />
+        </td>
+
+        <td className={styles.cell}>{row.typeLabel}</td>
+
+        <td className={styles.cell}>{row.location ?? '-'}</td>
+      </tr>
+
+      {isOpen && row.children && Object.keys(row.children).length > 0 && (
+        <NestedRows
+          rows={row.children}
+          selectedRows={selectedRows}
+          level={level + 1}
+          requestNestedRows={requestNestedRows}
+          onRowSelectedChange={onRowSelectedChange}
+        />
+      )}
+
+      {openStatus === 'loading' && (
+        <tr>
+          <td className={cx(styles.cell, styles.loadingCell)} colSpan={3}>
+            Loading...
+          </td>
+        </tr>
+      )}
+    </>
+  );
+};
+
+interface EntryIconProps {
+  entry: Row;
+  isOpen: boolean;
+}
+
+const EntryIcon: React.FC<EntryIconProps> = ({ isOpen, entry: { type } }) => {
+  switch (type) {
+    case EntryType.Collection:
+      return <Icon name="layer-group" />;
+
+    case EntryType.SubCollection:
+      return <Icon name={isOpen ? 'folder-open' : 'folder'} />;
+
+    case EntryType.Resource:
+      return <Icon name="cube" />;
+
+    default:
+      return null;
+  }
+};
+
+interface NestedEntryProps {
+  level: number;
+  entry: Row;
+  isSelected: boolean;
+  isOpen: boolean;
+  isDisabled: boolean;
+  onToggleCollapse: (row: Row) => void;
+  onSelectedChange: (row: Row, selected: boolean) => void;
+}
+
+const NestedEntry: React.FC<NestedEntryProps> = ({
+  entry,
+  isSelected,
+  isDisabled,
+  isOpen,
+  level,
+  onToggleCollapse,
+  onSelectedChange,
+}) => {
+  const theme = useTheme2();
+  const styles = useStyles2(getStyles);
+  const hasChildren = !!entry.children;
+  const isSelectable = entry.type === EntryType.Resource;
+
+  const handleToggleCollapse = useCallback(() => {
+    onToggleCollapse(entry);
+  }, [onToggleCollapse, entry]);
+
+  const handleSelectedChanged = useCallback(
+    (ev: React.ChangeEvent<HTMLInputElement>) => {
+      const isSelected = ev.target.checked;
+      onSelectedChange(entry, isSelected);
+    },
+    [entry, onSelectedChange]
+  );
+
+  return (
+    <div style={{ marginLeft: level * (3 * theme.spacing.gridSize) }}>
```

Comment:
it's a bit of a pain to to `css` here I think, and because its just indentation for the level (from props), and no other styles, i thought it would be fine?

Reply:
sure